src/components/filters/filter-container.component.ts

```diff
@@ -24,6 +24,9 @@ export class FilterContainerComponent implements OnDestroy {
     /** Emit when a specific event occurs */
     @Output() events = new EventEmitter<FilterEvent>();
 
+    /** Template ref for custom clear all functionality */
+    @Input() clearAll: TemplateRef<string>;
```

Comment:
Although we do have older components that work like this, we have mostly moved to providing templates as child `ng-template`s with a specific documented name, e.g.:
```html
<ux-filter-container ...>
  <ng-template #clearAllTemplate>
    Clear All
  </ng-template>
</ux-filter-container>
```

Check out the [Calendar](https://pages.github.houston.softwaregrp.net/caf/ux-aspects-micro-focus/#/ui-components/calendar) components for an example.

Reply:
Also, the type being used is incorrect. This is not `TemplateRef<string>`, this would be `TemplateRef<void>` as the generic represents the data passed to the template. You are not using `[ngTemplateOutletContext]` so you are not passing it any data, hence `void` would be the correct type.